Sources/PackageGraph/Pubgrub/PubgrubDependencyResolver.swift

```diff
@@ -927,7 +932,7 @@ private final class DiagnosticReportBuilder {
         if !positive.isEmpty && !negative.isEmpty {
             if positive.count == 1 {
                 let positiveTerm = terms.first{ $0.isPositive }!
-                return "\(description(for: positiveTerm, normalizeRange: true)) requires \(negative.joined(separator: " or "))";
+                return "\(description(for: positiveTerm, normalizeRange: true)) practically depends on \(negative.joined(separator: " or "))";
```

Comment:
Is the word "practically" necessary here?

Reply:
this one is interesting. it is used in two situations afaict:
1. when there is a transitive dependency 
2. when there is a constraint (tools version, etc) that "forces" the resolver to use a specific version(s).

I originally changed this to "transitively depends on" which did not fit the 2nd case, so changed it to "practically" which I dont love but I feel is better than the original "requires" as it at least tells you its deducted. totally open to better language tho

Reply:
I see.  Maybe "effectively", although I can't really say it's any more obvious than "practically".  We can always tweak it later, since these are just display strings and it shouldn't hold up the PR.